checker/src/org/checkerframework/checker/index/samelen/SameLenTransfer.java

```diff
@@ -73,16 +75,46 @@ public SameLenTransfer(CFAnalysis analysis) {
                                 UNKNOWN,
                                 arrayLengthNodeAnnotation);
 
-                result.getRegularStore().clearValue(targetRec);
-                result.getRegularStore().insertValue(targetRec, combinedSameLen);
-                result.getRegularStore().clearValue(otherRec);
-                result.getRegularStore().insertValue(otherRec, combinedSameLen);
+                propagateCombinedSameLen(combinedSameLen, node, result.getRegularStore());
             }
         }
 
+        // If the left side of the assignment has a SameLen type, propogate the right side to the arrays listed there.
+
+        AnnotationMirror leftAnno =
+                aTypeFactory
+                        .getAnnotatedType(node.getExpression().getTree())
+                        .getAnnotationInHierarchy(UNKNOWN);
+
+        if (leftAnno != null && AnnotationUtils.areSameByClass(leftAnno, SameLen.class)) {
+
+            AnnotationMirror combinedSameLen =
+                    aTypeFactory.createCombinedSameLen(
+                            node.getTarget().toString(),
```

Comment:
Instead of the Tree.toString(), the Receiver.toString should be used, because the Tree toString isn't standardized.  For example if the Tree is (this).field, the toString returns (this).field instead of this.field. Same goes for the next line.  Can you change the type of the parameters of createCombinedSameLen to Receiver to catch this problem sooner?